src/app/designationEditor/designationEditor.component.js

```diff
@@ -243,24 +314,45 @@ class DesignationEditorController {
   isPerson() {
     return this.designationContent && includes([
       'National Staff', 'Staff', 'Student', 'Volunteer', 'People'
-      ], this.designationContent.designationType);
+    ], this.designationContent.designationType);
   }
 
   isMinistry() {
     return this.designationContent && includes([
-        'Project', 'Scholarship', 'Ministry'
-      ], this.designationContent.designationType);
+      'Project', 'Scholarship', 'Ministry'
+    ], this.designationContent.designationType);
   }
 
   isCampaign() {
     return this.designationContent && includes([
-        'Campaign'
-      ], this.designationContent.designationType);
+      'Campaign'
+    ], this.designationContent.designationType);
+  }
+
+  getImageUrls(obj, imageUrls) {
```

Comment:
I'd prefer things to be more functional and immutable. I think it makes code much easier to follow. This is sorta shorthand, take whatever you want. I just think it's nicer to not have to pass the output array around and just have it return it.

```js
getImageUrls = obj =>
  Object.entries(obj).reduce(
    (acc, [key, value]) => [
      ...acc,
      ...(key === "fileReference"
        ? [value]
        : angular.isObject(obj)
        ? this.getImageUrls(value)
        : [])
    ],
    []
  );
```

Reply:
If you like what you have you could do this
```suggestion
  getImageUrls(obj, imageUrls = []) {
```
so you don't have to setup `imageUrls` beforehand. But I guess you'd still have to refactor to use return values.